gnomad_qc/v5/data_ingestion/federated_validity_checks.py

```diff
@@ -28,6 +28,70 @@
 logger.setLevel(logging.INFO)
 
 
+def validate_config(
+    ht: hl.Table, config_path: str, schema: Dict[str, Any]
+) -> Dict[str, Any]:
+    """Validate JSON config and check for missing fields in a Hail Table.
+
+    :param ht: Hail Table.
+    :param config_path: Path to the JSON configuration file.
+    :param schema: JSON schema to sue for validation.
+    :return: Config.
+    """
+    # Read in parameters from config file.
+    with hl.hadoop_open(config_path, "r") as f:
+        config = json.load(f)
+
+    # Validate config file against schema.
+    try:
+        validate(instance=config, schema=schema)
+        logger.info("JSON is valid.")
+    except ValidationError as e:
+        raise ValueError(f"JSON validation error: %s, {e.message}")
+
+    missingness_threshold = config["missingness_threshold"]
+    indexed_array_annotations = config["indexed_array_annotations"]
+    struct_annotations_for_missingness = config["struct_annotations_for_missingness"]
+    freq_meta_expr = eval(config["freq_meta_expr"])
+    freq_annotations_to_sum = config["freq_annotations_to_sum"]
+    freq_sort_order = config["freq_sort_order"]
+    nhomalt_metric = config["nhomalt_metric"]
```

Comment:
These are only used once, with the exception of  `indexed_array_annotations` and not returned. I think you could just directly reference the config dict in the code below instead of declaring these variables.